### PR TITLE
Convert documents to latest version before importing

### PR DESCRIPTION
**Pull request by fabubaker**
`import` now checks the version of documents to import and converts them to the latest version before running the 
import SQL.

This ensures that archived reviews can be restored to the latest version.

### CODE REVIEW

File: server/src/adapters/postgres/migrator/migrator.ts [hunk 1]
```diff
@@ -180,13 +180,38 @@ export class Migrator {
...
+  /**
+   * Converts an array of document objects to the specified version.
+   * @param documents - Array of document objects to be converted.
+   * @param target - The target table to import to.
+   * @param documentVersion - Current version of the document objects.
+   * @param convertVersion - Version to convert to.
+   */
+  async convertDocuments(documents: Array<any>, target: string, documentVersion: number, convertVersion: number): Promise<Array<any>> {
+    let convertedDocuments = documents
+    const versionsToConvert = _.filter(this.versions, (version) => {
```
Comment (hammady) [inline server/src/adapters/postgres/migrator/migrator.ts:202]: Just to confirm that `this.version` is already sorted and that the `_.filter` maintains the sort order.

File: server/src/adapters/postgres/migrations/00000000000000_create_initial_rayyan_store_tables.ts [hunk 2]
```diff
@@ -230,4 +230,8 @@ export class CreateInitialRayyanStoreTables implements BaseMigration {
   static async downgrade(reviewId: number, client: Pool|PoolClient): Promise<any> {
     return
   }
+
+  static async convertDocument(document: any, target: string): Promise<any> {
+    return
```
[the remark this answers is not in the document]
Comment (fabubaker) [reply server/src/adapters/postgres/migrations/00000000000000_create_initial_rayyan_store_tables.ts:235]: This is the base migration, so its converter never really gets called (what does it mean to apply a converter for the base migration anyways?).

File: server/src/adapters/postgres/migrator/migrator.ts (hunk 1 above)
Comment (fabubaker) [reply server/src/adapters/postgres/migrator/migrator.ts:202]: Yep:

```
this.versions = Object.keys(versionMappings).map(Number).sort()
```

File: server/src/adapters/postgres/migrations/00000000000000_create_initial_rayyan_store_tables.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (hammady) [reply server/src/adapters/postgres/migrations/00000000000000_create_initial_rayyan_store_tables.ts:235]: I understand, just for documentation purposes when someone looks at it as a reference migration.

**Review by hammady — APPROVED** [review]

**Pull request merged**